.github/workflows/continuous-integration.yml

```diff
@@ -24,31 +24,19 @@ jobs:
           - macos-latest
     steps:
       - uses: actions/checkout@v3
-
       - name: Set up Python ${{ matrix.python-version }}
         uses: actions/setup-python@v4
         with:
           python-version: ${{ matrix.python-version }}
           cache: "pip"
-          cache-dependency-path: |
-            setup.py
-            requirements-dev.txt
-
-      - name: Install package
-        run: pip install .
-
-      - name: Install dev requirements
-        run: pip install -r requirements-dev.txt
-
+      - name: Install package with dev requirements
+        run: pip install .[dev]
       - name: Run pre-commit
         run: pre-commit run --all-files
-
       - name: Run pytest
         run: pytest -Werror -s --block-network --cov pystac_client --cov-report term-missing
```

Comment:
Not sure if you want to include it in this PR, but it would be nice to move t least some of these pytest args to pyproject.toml